addons/mail/static/src/core/common/thread_model.js

```diff
@@ -759,14 +760,21 @@ export class Thread extends Record {
 
     /** @param {string} name */
     async rename(name) {
-        const newName = name.trim();
+        let newName = name.trim();
         if (
             newName !== this.displayName &&
             ((newName && this.channel_type === "channel") ||
                 this.channel_type === "chat" ||
                 this.channel_type === "group")
         ) {
             if (this.channel_type === "channel" || this.channel_type === "group") {
+                if (this.channel_type === "group" && !newName) {
+                    newName = formatList(
+                        this.channel_member_ids
+                            .slice(0, 4)
+                            .map((channelMember) => channelMember.persona.name)
+                    );
+                }
```

Comment:
Not all members might be known in JS, this should probably always be computed in python to avoid issues.

Reply:
I don't see the point of changing this though, if we call "rename" we expect the given name to be set, not to be called with an empty name.

Reply:
This is just preventing the group chat has an empty name. I did this so that we don't have to do an extra query in server side. But indeed, as you said, I would move it to the server side.